Mathlib/Analysis/BoxIntegral/Basic.lean

```diff
@@ -669,13 +669,186 @@ end Integrable
+  ⨅ S ∈ (𝓝 x).map f, EMetric.diam S
+
+/-- The oscillation of `f` at `x` is 0 whenever `f` is continuous at `x`. -/
+theorem oscillation_zero_of_continuousAt {f : ℝⁿ → E} {x : ℝⁿ} (hf : ContinuousAt f x) :
+    oscillation f x = 0 := by
+  refine le_antisymm (ENNReal.le_of_forall_pos_le_add fun ε hε _ ↦ ?_) (zero_le _)
+  rw [zero_add]
+  have : EMetric.ball (f x) (ε / 2) ∈ (𝓝 x).map f :=
+    hf <| EMetric.ball_mem_nhds _ (by simp [ne_of_gt hε])
+  refine (biInf_le EMetric.diam this).trans (le_of_le_of_eq diam_ball ?_)
+  refine (ENNReal.mul_div_cancel' ?_ ?_) <;> norm_num
+
+/-- If `oscillation f x < ε` at every `x` in a compact set `K`, then there exists `δ > 0` such
+that the oscillation of `f` on `Metric.ball x δ` is less than `ε` for every `x` in `K`. -/
+theorem uniform_oscillation_of_compact {K : Set ℝⁿ} (comp : IsCompact K) (f : ℝⁿ → E)
+    (ε : ENNReal) (hK : ∀ x ∈ K, oscillation f x < ε) :
+    ∃ δ > 0, ∀ x ∈ K, EMetric.diam (f '' (Metric.ball x δ)) ≤ ε := by
+  let S := fun r ↦ { x : ℝⁿ | ∃ (a : ℝ), (a > r ∧ EMetric.diam (f '' (Metric.ball x a)) ≤ ε) }
+  have S_open : ∀ r > 0, IsOpen (S r) := by
+    intro r _
+    rw [isOpen_iff_nhds]
+    rintro x ⟨a, ar, ha⟩ t ht
+    rw [Metric.mem_nhds_iff]
+    use (a - r) / 2, by simp [ar]
+    intro y hy
+    apply ht
+    use a - (a - r) / 2, by linarith
+    refine le_trans (diam_mono (Set.image_mono fun z hz ↦ ?_)) ha
+    rw [Metric.mem_ball] at *
+    linarith [dist_triangle z y x]
+  have S_cover : K ⊆ ⋃ r > 0, S r := by
+    intro x hx
+    have : oscillation f x < ε := hK x hx
+    simp only [oscillation, Filter.mem_map, iInf_lt_iff] at this
+    obtain ⟨n, hn₁, hn₂⟩ := this
+    obtain ⟨r, r0, hr⟩ := Metric.mem_nhds_iff.1 hn₁
+    use (S (r / 2)), ⟨r / 2, by simp [r0]⟩, r, div_two_lt_of_pos r0
+    exact le_trans (diam_mono (Set.image_subset_iff.2 hr)) (le_of_lt hn₂)
+  have S_antitone : ∀ (r₁ r₂ : ℝ), r₁ ≤ r₂ → S r₂ ⊆ S r₁ :=
+    fun r₁ r₂ hr x ⟨a, ar₂, ha⟩ ↦ ⟨a, lt_of_le_of_lt hr ar₂, ha⟩
+  have : ∃ r > 0, K ⊆ S r := by
+    obtain ⟨T, Tb, Tfin, hT⟩ := comp.elim_finite_subcover_image S_open S_cover
+    by_cases T_nonempty : T.Nonempty
+    · use Tfin.isWF.min T_nonempty, Tb (Tfin.isWF.min_mem T_nonempty)
+      intro x hx
+      obtain ⟨r, hr⟩ := Set.mem_iUnion.1 (hT hx)
+      simp only [Set.mem_iUnion, exists_prop] at hr
+      exact (S_antitone _ r (Set.IsWF.min_le Tfin.isWF T_nonempty hr.1)) hr.2
+    · rw [Set.not_nonempty_iff_eq_empty] at T_nonempty
+      use 1, one_pos, subset_trans hT (by simp [T_nonempty])
+  obtain ⟨δ, δ0, hδ⟩ := this
+  use δ, δ0
+  intro x xK
+  obtain ⟨a, δa, ha⟩ := hδ xK
+  exact le_trans (diam_mono (Set.image_mono (Metric.ball_subset_ball (le_of_lt δa)))) ha
+
+end oscillation
+
 /-!
 ### Integrability conditions
 -/
 
+open Prepartition EMetric
 
 variable (l)
 
+/-- A function that is bounded and a.e. continuous on a box `I` is integrable on `I`. -/
+theorem integrable_of_bounded_and_ae_continuous (l : IntegrationParams) [CompleteSpace E]
```

Comment:
Can you make sure the arguments are in the same order as in `integrable_of_continuousOn`? I don't think it's the case at the moment.

Reply:
I think the only change in the order is that now hc is introduced after mu because it depends on mu.